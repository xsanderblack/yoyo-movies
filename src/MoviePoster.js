import React, { Component } from 'react';
import posterThumbnailSize from './images/poster-thumbnail-size.jpg';
import posterFullSize from './images/poster-full-size.jpg';
import {
  POSTER_PATH_BASE,
  POSTER_THUMBNAIL_SIZE,
  POSTER_FULL_SIZE
} from './Constants';

// Component returns poster image in one of two sizes,
// and replases broken links with default image.
export default class MoviePoster extends Component {
  addDefaultImage = (e) => {
    const { size } = this.props;

    if (size === 'thumbnail') {
      e.target.src = posterThumbnailSize;
    } else if (size === 'full') {
      e.target.src = posterFullSize;
    }
  };

  render() {
    const { size, path, title } = this.props;
    let posterSize;

    if (size === 'thumbnail') {
      posterSize = POSTER_THUMBNAIL_SIZE;
    } else if (size === 'full') {
      posterSize = POSTER_FULL_SIZE;
    }

    return (
      <img
        src={`${POSTER_PATH_BASE}${posterSize}${path}`}
        alt={title}
        onError={this.addDefaultImage}
      />
    );
  };
};
