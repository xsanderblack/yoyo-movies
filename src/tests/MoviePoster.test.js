import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import MoviePoster from '../MoviePoster';

it('MoviePoster renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MoviePoster />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('should render MoviePoster correctly', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<MoviePoster />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
