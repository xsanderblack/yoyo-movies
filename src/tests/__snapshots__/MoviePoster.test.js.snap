// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MoviePoster correctly 1`] = `
<img
  onError={[Function]}
  src="https://image.tmdb.org/t/p/undefinedundefined"
/>
`;
